### PR TITLE
Use new sdp again

### PR DESCRIPTION
**Pull request by MeouSker77**
## Description

change chatglm2/3 and baichuan2 to use new sdp, and check `head_dim` in `use_sdp_causal`

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/models/baichuan2.py [hunk 1]
```diff
@@ -166,9 +167,8 @@ def baichuan_attention_forward_7b_quantized(
 
     past_key_value = (key_states, value_states) if use_cache else None
 
-    if attention_mask is not None:
-        if attention_mask.dtype == torch.bool:
-            attention_mask.masked_fill_(attention_mask.logical_not(), float("-inf"))
+    invalidInputError(attention_mask is None or attention_mask.dtype != torch.bool,
```
Comment (rnwang04) [inline python/llm/src/ipex_llm/transformers/models/baichuan2.py:170]: I think we can manually turn attention_mask from bool into float ?

**Review by rnwang04 — APPROVED** [review]

File: python/llm/src/ipex_llm/transformers/models/baichuan2.py (hunk 1 above)
Comment (MeouSker77) [reply python/llm/src/ipex_llm/transformers/models/baichuan2.py:170]: baichuan2-7b is using float mask, it doesn't use `bool` mask
